### PR TITLE
Update user guide

### PR DESCRIPTION
**Pull request by xyT-T**
Update user guide to include delete task, mark and unmark commands.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -288,6 +289,51 @@ Format: `add_task td/TASK_DESCRIPTION d/DEADLINE ev/EVENT_NAME`
...
+* Both the `TASK_DESCRIPTION` and the `EVENT_NAME` are case-insensitive.
+ e.g. `delete_task td/Book Venue ev/NUS Career Fair 2023` and `delete_task td/book venue ev/nus Career FAIR 2023`
+ will perform the same operation.
+* Warnings will be sent if the specified event does not exist or the event does not have the specified task.
+ If such situation happens, you may double-check the task description and the event name and reenter valid inputs.
+
+Examples:
+* `delete_task td/Book Venue ev/NUS Career Fair 2023` deletes task `Book Venue` from the task list of the event `NUS Career Fair 2023`.
+
+### Marking a task: `mark`
```
Comment (vijay-shankaranand) [inline docs/UserGuide.md:308]: This should be mark_task

File: docs/UserGuide.md [hunk 2]
```diff
@@ -288,6 +289,51 @@ Format: `add_task td/TASK_DESCRIPTION d/DEADLINE ev/EVENT_NAME`
...
+* Both the `TASK_DESCRIPTION` and the `EVENT_NAME` are case-insensitive.
+  e.g. `mark td/Book Venue ev/NUS Career Fair 2023` and `mark td/book venue ev/nus Career FAIR 2023`
+  will perform the same operation.
+* Warnings will be sent if the specified event does not exist or the event does not have the specified task.
+  If such situation happens, you may double-check the task description and the event name and reenter valid inputs.
+
+Examples:
+* `mark td/Book Venue ev/NUS Career Fair 2023` marks the task `Book Venue` from the task list of the event `NUS Career Fair 2023` as completed.
+
+### Unmarking a task: `unmark`
```
Comment (vijay-shankaranand) [inline docs/UserGuide.md:325]: This should be unmark_task

File: docs/UserGuide.md [hunk 3]
```diff
@@ -288,6 +289,51 @@ Format: `add_task td/TASK_DESCRIPTION d/DEADLINE ev/EVENT_NAME`
...
+
+Deletes the task specified by the task description from its associated event in JobFestGo.
+
+Format: `delete_task td/TASK_DESCRIPTION ev/EVENT_NAME`
+
+* Deletes the task specified by the `TASK_DESCRIPTION` from the event specified by the `EVENT_NAME` of JobFestGo.
+* Both the `TASK_DESCRIPTION` and the `EVENT_NAME` are case-insensitive.
+ e.g. `delete_task td/Book Venue ev/NUS Career Fair 2023` and `delete_task td/book venue ev/nus Career FAIR 2023`
+ will perform the same operation.
+* Warnings will be sent if the specified event does not exist or the event does not have the specified task.
```
Comment (vijay-shankaranand) [inline docs/UserGuide.md:302]: should this be this instead
```suggestion
* Error will be sent if the specified event does not exist or the event does not have the specified task.
```

File: docs/UserGuide.md [hunk 4]
```diff
@@ -288,6 +289,51 @@ Format: `add_task td/TASK_DESCRIPTION d/DEADLINE ev/EVENT_NAME`
...
+Marks the task specified by the task description and its associated event name in JobFestGo as completed.
+
+Format: `mark td/TASK_DESCRIPTION ev/EVENT_NAME`
+
+* Marks the task specified by the `TASK_DESCRIPTION` from the event specified by the `EVENT_NAME` of JobFestGo as completed.
+ A label indicating the status of the task will be shown next to its description in the task list.
+* Both the `TASK_DESCRIPTION` and the `EVENT_NAME` are case-insensitive.
+  e.g. `mark td/Book Venue ev/NUS Career Fair 2023` and `mark td/book venue ev/nus Career FAIR 2023`
+  will perform the same operation.
+* Warnings will be sent if the specified event does not exist or the event does not have the specified task.
```
Comment (vijay-shankaranand) [inline docs/UserGuide.md:319]:
```suggestion
* Error will be sent if the specified event does not exist or the event does not have the specified task.
```

**Review by vijay-shankaranand — CHANGES_REQUESTED** [review]
Needs some critical fix, please take a look and feel free to clarify if needed!

File: docs/UserGuide.md (hunk 3 above)
Comment (vijay-shankaranand) [reply docs/UserGuide.md:302]: I feel that error is different from a warning. A warning would mean that the system still accepts the input but reminds the user that something can go wrong, whereas an error outright prevents the input, similar to how checkstyle warnings vs errors work

Comment (xyT-T) [reply docs/UserGuide.md:302]: I think this is a good suggestion! Thank you!

File: docs/UserGuide.md
Comment (rionshocker) [inline docs/UserGuide.md:1]: Missing command summary for `delete_task`, `mark_task`, and `unmark_task`

**Review by rionshocker — CHANGES_REQUESTED** [review]
Apart from the missing command summary, the rest looks good to me!

File: docs/UserGuide.md
Comment (xyT-T) [reply docs/UserGuide.md:1]: Thanks for reminding! I've added them in.

**Review by vijay-shankaranand — APPROVED** [review]
LGTM!

**Pull request merged**